tests/integration/20.command.functions.test.js

```diff
@@ -840,4 +840,25 @@ test('should handle content-types with charset', async (t) => {
   })
 })
 
+test('should execute function when name clashes with config key names', async (t) => {
```

Comment:
The fix itself looks good to me, but I'm not sure this is the right test to add. The fact that a function name was clashing with a config property was not the underlying problem, it was one of the side effects. The underlying problem was that CLI was sending the wrong configuration object to zip-it-and-ship-it, and I think that's what we should assert in the test.

I think the best way to do that would be to stub the call to zip-it-and-ship-it, which isn't very straightforward in this type of integration test (i.e. using `withDevServer` and testing the end-to-end flow).

Instead, you could write a unit test for the functions registry, asserting that it receives the entire configuration in its constructor but then passes just the functions block to zip-it-and-ship-it.